balancer-js/src/modules/pools/index.ts

```diff
@@ -43,9 +43,9 @@ export class Pools {
           slippage,
           wrappedNativeAsset: networkConfig.addresses.tokens.wrappedNativeAsset,
         }),
-      apr: async function () {
+      fetchApr: async function () {
         const aprService = new PoolApr(
-          data,
+          this,
```

Comment:
The reason we need to pass `this` is so that if the pool is updated and `apr()` is called it uses the updated data. When just extending that data object it creates a copy of it and the apr function uses a separate copy of the old data.